src/Crypto/Spake2.hs

```diff
@@ -140,15 +134,10 @@ newtype SideID = SideID { unSideID :: ByteString } deriving (Eq, Ord, Show)
 -- | Convert a user-supplied password into a scalar on a group.
 passwordToScalar :: Group group => group -> Password -> Scalar group
 passwordToScalar group password =
-  let oversized = expandPassword password (scalarSizeBytes group + 16) :: ByteString
-  in decodeScalar group oversized
-
--- | Expand a password using HKDF so that it has a certain number of bytes.
---
--- TODO: jml cannot remember why you might want to call this.
-expandPassword :: ByteArray output => Password -> Int -> output
-expandPassword (Password bytes) numBytes = expandData info bytes numBytes
+  decodeScalar group oversized
   where
+    oversized = expandPassword password (scalarSizeBytes group + 16) :: ByteString
+    expandPassword (Password bytes) = expandData info bytes
```

Comment:
Exactly right on all counts. It's a refactoring to reduce the scope.

This is called "point-free style", so called because the definition of the function doesn't mention the input 'point'.

Broadly speaking, you could write:

```haskell
sum xs = foldr (+) xs
``` 

as

```haskell
sum = foldr (+)
```

I've mostly avoided it in this code-base up until now, precisely because it is confusing to newcomers (and to many people who aren't newcomers, including SPJ himself). Unfortunately, I have a linter that keeps suggesting point-free style to me, and I occasionally reflexively do what it says.